presto-main/src/main/java/com/facebook/presto/sql/analyzer/TupleAnalyzer.java

```diff
@@ -531,35 +530,34 @@ protected TupleDescriptor visitValues(Values node, AnalysisContext context)
     {
         checkState(node.getRows().size() >= 1);
 
-        TupleAnalyzer analyzer = new TupleAnalyzer(analysis, session, metadata, sqlParser, experimentalSyntaxEnabled);
+        Set<Type> types = node.getRows().stream()
+                .map((row) -> analyzeExpression(row, new TupleDescriptor(), context).getType(row))
+                .collect(ImmutableCollectors.toImmutableSet());
 
-        // Use the first descriptor as the output descriptor for the VALUES
-        TupleDescriptor outputDescriptor = analyzer.process(node.getRows().get(0), context).withOnlyVisibleFields();
-        Iterable<Type> types = transform(outputDescriptor.getVisibleFields(), Field::getType);
-
-        for (Row row : Iterables.skip(node.getRows(), 1)) {
-            TupleDescriptor descriptor = analyzer.process(row, context);
-            Iterable<Type> rowTypes = transform(descriptor.getVisibleFields(), Field::getType);
-            if (!elementsEqual(types, rowTypes)) {
-                throw new SemanticException(MISMATCHED_SET_COLUMN_TYPES, node, "Values rows have mismatched types: " +
-                        "Expected: (" + Joiner.on(", ").join(types) + "), " +
-                        "Actual: (" + Joiner.on(", ").join(rowTypes) + ")");
-            }
+        if (types.size() > 1) {
+            throw new SemanticException(MISMATCHED_SET_COLUMN_TYPES,
```

Comment:
They are not, just like in the current implementation. I filed an issue to track this: https://github.com/facebook/presto/issues/2198